app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -69,6 +69,7 @@ def import_all(uuids = nil, options = {})
 
       def import_for_repository(repository, force = false)
         ids = content_unit_class.ids_for_repository(repository.pulp_id)
+        ids -= repository.rpms.map(&:uuid) if self == Katello::Rpm
```

Comment:
We could do for errata and package groups too, rpms will have the largest benefit by far. @jlsherrill did you have concerns about adding errata and package groups to this logic as well?